ratpack-exec/src/main/java/ratpack/exec/Operation.java

```diff
@@ -86,7 +87,38 @@ static Operation flatten(Factory<Operation> factory) {
     return new DefaultOperation(Promise.flatten(() -> factory.create().promise()));
   }
 
-  Operation onError(Action<? super Throwable> onError);
+  default Operation onError(Action<? super Throwable> onError) {
+    return onError(Predicate.TRUE, onError);
+  }
+
+  /**
+   * Specifies the action to take if the an error occurs performing the operation that the given predicate applies to.
+   * <p>
+   * If the given action throws an exception, the original exception will be rethrown with the exception thrown
+   * by the action added to the suppressed exceptions list.
+   *
+   * @param predicate the predicate to test against the error
+   * @param errorHandler the action to take if an error occurs
+   * @return An operation for the successful result
+   * @since 1.8.3
```

Comment:
Presumptuous 😜

Reply:
public API has to be added in a minor release. Luckily....we already have a 1.9.0 in the works.
Can you update this comment and target the PR to the 1.9.x branch?

Reply:
Done 😄 

Cheers John 👍